### PR TITLE
Ipxe rollup

### PR DESCRIPTION
**Pull request by jlevon**

### CODE REVIEW

**Comment by jlevon** [comment]
This PR contains fixes for:

OS-7748 iPXE EFI boot can fail if bounce buffer space is not available
OS-7782 iPXE EFI ljmp fails under KVM Linux
OS-8138 iPXE EFI boot under vmware stumbles over early allocations
TRITON-2097 update ipxe build to 19.4.0 build image

and goes hand in hand with:

https://github.com/joyent/illumos-joyent/pull/269

File: src/arch/x86/image/multiboot2_entry.S [hunk 1]
```diff
@@ -61,8 +61,22 @@ multiboot2_entry:
...
 	lgdt	(%rax)
 
-	/* Load our new %cs. */
-	ljmp	*newcs_vector
+	/*
+	 * Load our new %cs.
+	 *
+	 * Note that originally, this was an ljmp; while this worked on many
+	 * systems, it appeared to have issues on at least VirtualBox (see
+	 * https://www.virtualbox.org/ticket/19414), so we set up on iretq frame
```
Comment (danmcd) [inline src/arch/x86/image/multiboot2_entry.S]: Grammar nit:  "on iretq frame"?  Or "an iretq frame"?

Comment (danmcd) [reply src/arch/x86/image/multiboot2_entry.S]: Also, did this fix the world in Linux KVM too? Testing info on ticket suggests it might've.

**Review by danmcd — COMMENTED** [review]
Nits...

File: src/arch/x86/image/multiboot2_entry.S (hunk 1 above)
Comment (jlevon) [reply src/arch/x86/image/multiboot2_entry.S]: done

**Review by danmcd — APPROVED** [review]
Thank you!

**Pull request merged**